### ISSUE TITLE
system shutdown takes an unreasonably longer time to happen on Linux Mint 18.3 with Steam Beta

### ISSUE THREAD

**Issue by paulovilli** [ValveSoftware/steam-for-linux#5379]
#### Your system information

* Steam client version (build number or date): jan 25 2018 at 21:25:07 / 1516948201 / API v018
* Distribution (e.g. Ubuntu): Linux Mint Cinnamon Edition 18.3
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes
* Hardware: Phenom-II x4 + AMD HD7770
* Linux Kernel: 4.13.0-32 (Ubuntu Kernel HWE, as provided by Linux Mint Update Manager)
(also tested on kernel 4.4.0-112, also Ubuntu Kernel provided by Linux Mint Update Manager)
* Steam installed from Linux Mint's Software Manager repository
* GPU Driver: opensource
* CPU Microcode Update: amd64-microcode (version 2.20160316.1)
* PPAs and other custom repositories: none of relevance
* More detailed system info: https://gist.github.com/anonymous/1feaa7063988e021d2b3b8c196f10846

#### Please describe your issue in as much detail as possible:
I had the issue described here, which was very recently fixed on Steam Beta:
https://github.com/ValveSoftware/steam-for-linux/issues/3301

Enabling Beta resolved the original issue, but now shutdown takes an unreasonably longer time to happen on Linux Mint 18.3, as evidenced on these coments to the original issue:
- https://github.com/ValveSoftware/steam-for-linux/issues/3301#issuecomment-360739078
- https://github.com/ValveSoftware/steam-for-linux/issues/3301#issuecomment-360912792
- https://github.com/ValveSoftware/steam-for-linux/issues/3301#issuecomment-360985271
- https://github.com/ValveSoftware/steam-for-linux/issues/3301#issuecomment-360993189
- https://github.com/ValveSoftware/steam-for-linux/issues/3301#issuecomment-362969407

Before switching from normal Steam app to the latest beta I have measured shutdown time as this:

> First shutdown attempt took 2sec to close Steam (tray icon disappearing) and second attempt issued 1sec later took 6sec to power off the PC.

Now, on current beta:

> I timed it to 38 seconds now, of which only 5 are before Steam's tray icon disappears and almost all of it is staring at the desktop with barely no signs that the shutdown sequence is still running.

New issue was opened because the sum of the time needed to close Steam and then Linux Mint shutdown before the fix was much smaller (even with human interaction) than now when both actions happen in a direct sequence with no other human interaction after requesting shutdown:
> So even counting the time lost with a secondary human interaction it took a lot less time to shutdown before the beta fix.

Of note: so far, only Linux Mint Cinnamon 18.3 reported such long delays. Other distros affected by original issue seem unaffected and even a user of Linux Mint **KDE** 18.2 noticed longer times but not comparable to mine and @Zerophase as evidenced here:
https://github.com/ValveSoftware/steam-for-linux/issues/3301#issuecomment-362939622

#### Steps for reproducing this issue:

1. Install Steam on Linux Mint Cinnamon 18.3
2. Login on Steam
3. Start cronometer, ask the system to shutdown/reboot, wait for the Steam tray icon to disappear, ask again for shutdown/reboot, wait until system is powered off or reboots, stop cronometer
4. Enable Steam Beta
5. Repeat step 3 (except no secondary shutdown request is necessary because original bug is fixed)
6. Compare results

**Comment by mlewis-everley** [comment]
Yep, I am getting this too, both on Mint 17 and 18 (both using Cinnamon).

**Comment by Seil0** [comment]
I am getting this too on opensuse tumbleweed with kde 5.12.1

**Comment by stevethepocket** [comment]
For what it's worth, this isn't happening for me on Kubuntu 17.10, with KDE Plasma 5.10.5. Shutdown happens almost instantly, faster than in Windows most of the time.

**Comment by dajusc** [comment]
I am still having this issue on Linux Mint 19 Cinnamon.

**Comment by breadland** [comment]
I can confirm that not only does this problem still persist on Linux Mint 19.1 but also on Arch Linux with the Cinnamon DE.

Any other DE, like XFCE, KDE, Gnome, and Deepin don't have this problem. I have tested these DE's on Arch, and have specifically tested stock Ubuntu 18.10, Ubuntu 19.04, Kubuntu 18.10, Xubuntu 18.10, KDE Neon, Deepin OS, and Pop! OS 19.04 without this issue.

On Linux Mint/Cinnamon DE, it takes exactly 30 seconds to shutdown/logout from when I click the button.

**Comment by Timeroot** [comment]
Having this issue as well. Almost exactly the same as #3301. Happens on both the Steam Client Beta, and not (on the latest versions available today, May 8th 2020).

This is on Ubuntu 18.04 with a Cinnamon Desktop Environment.
